### ISSUE TITLE
Add `access site-wide contact form` permission to anonymous by default

### ISSUE THREAD

**Issue by opi** [backdrop/backdrop-issues#3222]
This is a spun off from #1168. Part of [[META] Contact module improvements #3023](https://github.com/backdrop/backdrop-issues/issues/3023).

> Hit that page as anonymous... access denied! WTF!? OK, drupal old-timer goes to permission page and sets permissions for anonymous. All good.

When enabling the contact module, I guess I'm not the only one who wants the contact form to be available to everybody.

If one don't, note that the permissions will be available on the contact settings page: https://github.com/backdrop/backdrop-issues/issues/3190

**Comment by opi** [comment]
Actually, permissions by role are never defined by module themselves. This is done during the profile installation step. 
As the contact module is not in the minimal nor the standard profile, maybe it's a wrong idea to call `user_role_grant_permissions` during the `hook_install`.

Another suggestion will be to display an **info message** (once #3155 is merged) to tell the site administrator that permissions can be defined in the contact settings page.

**Comment by klonos** [comment]
Yes, I guess that this is a decision of simplicity (good site builder UX) vs playing it safe. Ideally, when everything comes together, upon enabling the Contact module, there will be a new message of type `info` (#3155) that will be explaining that the Contact module by default allows anonymous sit visitors to access the form. The same link can have a link to the contact from settings page, where one could harden the security if that is what they want. Over time, once we have #285 in place, we will be able to make better decision whether that default should be allow or block anonymous site visitors.
